src/pvrclient-mythtv.cpp

```diff
@@ -2581,22 +2581,27 @@ void PVRClientMythTV::FillRecordingAVInfo(MythProgramInfo& programInfo, Myth::St
 
 time_t PVRClientMythTV::GetRecordingTime(time_t airtt, time_t recordingtt)
 {
-  if (!g_bUseAirdate || airtt == 0)
-    return recordingtt;
+  // NOTE: Kodi wants everything in local time, MythTV has everything in UTC
+  struct tm rectm;
+  localtime_r(&recordingtt, &rectm);
 
   /* Airdate is usually a Date, not a time.  So we include the time part from
   the recording time in order to give the reported time something other than
   12AM.  If two shows are recorded on the same day, typically they are aired
   in the correct time order.  Combining airdate and recording time gives us
   the best possible time to report to the user to allow them to sort by
   datetime to see the correct episode ordering. */
-  struct tm airtm, rectm;
-  gmtime_r(&airtt, &airtm);
```

Comment:
I think the initial issue was to use gmtime + mktime , which isn't regular. But I don't know exactly the need. "gmtime" loose the DLS and fill the struct with UTC time. The result tm isn't usable with "mktime" for a regular operation. Maybe your issue was here ?